lib/restforce/concerns/streaming.rb

```diff
@@ -30,6 +31,65 @@ def faye
           client.bind 'transport:up' do
             Restforce.log "[COMETD UP]"
           end
+
+          client.add_extension ReplayExtension.new(replay_handlers)
+        end
+      end
+
+      def replay_handlers
+        @_replay_handlers ||= {}
+      end
+
+      class ReplayExtension
+        def initialize(replay_handlers)
+          @replay_handlers = replay_handlers
+        end
+
+        def incoming(message, callback)
+          callback.call(message).tap do
+            channel = message.fetch('channel').gsub('/topic/', '')
+            replay_id = message.fetch('data', {}).fetch('event', {})['replayId']
+
+            handler = @replay_handlers[channel]
+            if !replay_id.nil? && !handler.nil? && handler.respond_to?(:[]=)
```

Comment:
@georges I see what you mean. 

When I wrote it, I had in mind that if you pass a scalar it will not be updated ever. The idea being that I want to specify one of the magic values for this subscription:

* -1  Subscriber receives new events that are broadcast after the client subscribes.
* -2  Subscriber receives all events, including past events that are within the 24-hour retention window and new events sent after subscription.

It was my intent to have hashes cover the case where you want to track the latest values. Because we're mutating the replay value, it can't be a scalar.

I admit that that isn't immediately obvious, and I'm not altogether happy with the reference semantics involved with this implementation anyway. Meaning I don't really like that you pass an object to the subscription that you expect it to mutate for you.

All that said I think this could work as is, even if it's not ideal. perhaps we can replace -1 and -2 with :new_events and :past_events or similar?

Thoughts?